chrome/sunset.js

```diff
@@ -163,12 +182,59 @@ Sunset.showArticle_ = function() {
 
 
 /**
+ * Runs periodically to check if the extension has been running for another
+ * day. If so, it decreases the TTL. When TTL reaches zero, the extension
+ * is automatically uninstalled.
+ * @private
+ */
+Sunset.heartbeat_ = function() {
+  // If we're past the hard deadline, just uninstall the extension.
+  var today = new Date();
+  today.setHours(0, 0, 0, 0);
+  if (today >= Sunset.deadline_) {
+    chrome.management.uninstallSelf();
+    return;
+  }
+
+  // Otherwise, check if a whole day has passed and decrease the TTL.
+  chrome.storage.sync.get(null, function(data) {
+    // Read the remaining TTL and last uptime date from the storage.
+    var ttl = parseInt(data.ttl, 10);
+    var last_date = new Date(data.date);
+
+    if (isNaN(ttl) || ttl >= Sunset.max_uptime_)
+      ttl = Sunset.max_uptime_;
+
+    if (!last_date.getTime())
+      last_date = new Date();
+
+    // If a day has passed, decrease the TTL. When TTL reaches zero,
+    // uninstall the extension.
+    last_date.setHours(0, 0, 0, 0);
+    if (today > last_date)
+      --ttl;
+
+    if (ttl <= 0) {
+      chrome.management.uninstallSelf();
+      return;
+    }
+
+    // Write the new TTL and date.
+    chrome.storage.sync.set({
+      "ttl": ttl,
+      "date": today.toString()
+    });
+  });
+}
+
+
+/**
  * Triggers a loop that tests if conditions were met to show the message.
  */
 Sunset.run = function() {
   chrome.browserAction.onClicked.addListener(Sunset.showArticle_);
   chrome.notifications.onButtonClicked.addListener(Sunset.showLandingPage_);
-  chrome.alarms.onAlarm.addListener(Sunset.maybeShowNotification_);
```

Comment:
Yes, and several other methods and attributes that were only referenced by it (I wanted to do a large cleanup CL in the end, but we can also do this now).